### PR TITLE
Use sbt makeScalaInstance

### PR DESCRIPTION
**Pull request by adpi2**
and stop using sbt private `extendedClassLoaderCache`

### CODE REVIEW

File: project/Bootstrap.scala [hunk 1]
```diff
@@ -1,55 +0,0 @@
...
-
-  /** A ScalaInstance identical to `base` but with additional jars for scaladoc. */
-  def makeDocScalaInstance(
-    state: State,
-    base: ScalaInstance,
-    docJars: Array[File]
-  ): ScalaInstance = {
-    val cache = state.extendedClassLoaderCache
-
-    val fullLoader = cache(docJars.toList, base.loaderCompilerOnly)
```
Comment (smarter) [inline project/Bootstrap.scala:42]: This version reused the classloader from base for compiler and library jars, whereas the new one will recreate it it seems, isn't that likely to be slower and require more memory?

Comment (adpi2) [reply project/Bootstrap.scala:42]: Class loaders are cached internally by sbt, based on the jars file. Here we use compiler jars from the base Scala Instance, so the compiler class loader should be the same instance.

**Review by smarter — APPROVED** [review]

File: project/Bootstrap.scala (hunk 1 above)
Comment (adpi2) [reply project/Bootstrap.scala:42]: Actually there is something wrong the list of compiler jars, so I will fix it and add an assert to ensure that the class loaders are the same.

Comment (adpi2) [reply project/Bootstrap.scala:42]: The problem is https://github.com/sbt/sbt/blob/fabeed222799efcc46be15bccf2c31e384f7e369/main/src/main/scala/sbt/Defaults.scala#L1186-L1187

It's like a snake biting it's own tail...

**Pull request merged**